locals.tf

```diff
@@ -7,7 +7,7 @@ locals {
   aws_caller_identity_user_name = element(split("/", data.aws_caller_identity.terraform.arn), length(split("/", data.aws_caller_identity.terraform.arn)) - 1)
 
   # General
-  sec_group                 = coalescelist(aws_security_group.sg_a, aws_security_group.sg_b)
+  sec_group                 = (length(aws_security_group.sg_a) == 0 && length(aws_security_group.sg_b) == 0) ? null : coalescelist(aws_security_group.sg_a, aws_security_group.sg_b)
```

Comment:
you'll run into issues here as length requires a length or a valid object if the object is null or can be set to null anywhere you'll need to do something like a can() on the items.

Reply:
@thpang , for now aws_security_groups sg_a and sg_b are security group resources that will either have a valid non zero length if either one of them is defined or be an empty tuple of length zero when their count expressions evaluate to zero (can be true for both at once). Those objects are not being set anywhere else or to null at present so expect what I have is safe now for each of the tested scenarios. Does that mitigate the concern or need to use can() or try()? I looked at both and can consider using them if you recommend it.